### PR TITLE
Enable HIP static build option for image_augmentations

### PR DESCRIPTION
**Pull request by r-abishek**

### CODE REVIEW

File: src/modules/hip/kernel/blend.cpp [hunk 1]
```diff
@@ -1,67 +1,115 @@
...
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= width || id_y >= height || id_z >= channel)
+    {
+        return;
+    }
 
     int pixIdx = id_x + id_y * width + id_z * width * height;
 
     output[pixIdx] = ((1-alpha) * input1[pixIdx]) + (alpha * input2[pixIdx]);
```
Comment (rrawther) [inline src/modules/hip/kernel/blend.cpp:33]: need to rewrite this taking advantage of float vector computation

File: src/modules/hip/kernel/brightness.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    float alphatmp = alpha[id_z], betatmp = beta[id_z];
+    long pixIdx = 0;
+
+    pixIdx = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * plnpkdindex;
+
+    if((id_y >= yroi_begin[id_z]) && (id_y <= yroi_end[id_z]) && (id_x >= xroi_begin[id_z]) && (id_x <= xroi_end[id_z]))
+    {
+        for(int indextmp = 0; indextmp < channel; indextmp++)
```
Comment (rrawther) [inline src/modules/hip/kernel/brightness.cpp:68]: do float3 computation and avoid channel loop

File: src/modules/hip/kernel/fog.cpp [hunk 3]
```diff
@@ -1,90 +1,103 @@
...
+        return;
+    }
+
+    int pixId= width * id_y + id_x;
+    int c = width * height;
+    float check = input[pixId];
+    if(channel > 1)
+    {
+        check += input[pixId+c] + input[pixId + c * 2];
+        check = check / 3;
```
Comment (rrawther) [inline src/modules/hip/kernel/fog.cpp:30]: please avoid this division. replace by float multiply

File: src/modules/hip/kernel/fog.cpp [hunk 4]
```diff
@@ -1,90 +1,103 @@
...
     }
     if(check >= (240) && fogValue!=0)
     {}
     else if(check>=(170))
     {
-        float pixel = ((float) input[pixId])  * (1.5 + fogValue) - (fogValue*4) + (7*fogValue);
+        float pixel = ((float) input[pixId]) * (1.5 + fogValue) - (fogValue*4) + (7 * fogValue);
```
Comment (rrawther) [inline src/modules/hip/kernel/fog.cpp:36]: need to compute all 3 values at one time using vector instructions

File: src/modules/hip/kernel/snow.cpp [hunk 5]
```diff
@@ -102,58 +130,65 @@ extern "C" __global__ void snow_pln( unsigned char *output,
                 if (id_x + i + 5 <= width && id_y + j < height)
```
Comment (rrawther) [inline src/modules/hip/kernel/snow.cpp:130]: this looks like a bug. I think id_x+i <= width is good. Please double check

File: src/modules/hip/kernel/snow.cpp [hunk 6]
```diff
@@ -102,58 +130,65 @@ extern "C" __global__ void snow_pln( unsigned char *output,
...
     if(id_x < width[id_z] && id_y < height[id_z])
     {
-        pixIdx = batch_index[id_z] + (id_x  + id_y * max_width[id_z] ) * plnpkdindex ;
+        pixIdx = batch_index[id_z] + (id_x + id_y * max_width[id_z]) * plnpkdindex;
         if((id_y >= 0 ) && (id_y < height[id_z]) && (id_x >= 0) && (id_x < width[id_z]))
-        {   
+        {
             float pixelDistance = 1.0;
-            pixelDistance /=  (rainProbTemp / 100);
+            pixelDistance /= (rainProbTemp / 100);
```
Comment (rrawther) [inline src/modules/hip/kernel/snow.cpp:171]: avoid this division in the low level kernel. Please pass float parameter from high level

**Review by rrawther — CHANGES_REQUESTED** [review]

File: src/include/hip/rpp/hip_kernels.h [hunk 7]
```diff
@@ -1072,7 +1063,7 @@ unsigned char blend_formula(unsigned char input_pixel1,unsigned char input_pixel
 
 // extern "C" __global__ void brightness_contrast_batch(unsigned char *input,unsigned char *output,float *alpha,float *beta,int *xroi_begin,int *xroi_end,int *yroi_begin,int *yroi_end,unsigned int *height,unsigned int *width,unsigned int *max_width,unsigned long *batch_index,const unsigned int channel,unsigned int *inc,const int plnpkdindex);
 // extern "C" __global__ void brightness_contrast_ROI(unsigned char *input,unsigned char *output,const float alpha,const float beta,const int x1,const int y1,const int roi_height,const int roi_width,const unsigned int batch_index,const unsigned int height,const unsigned int width,const unsigned int channel,const ushort pln);
-extern "C" __global__ void brightness_contrast(unsigned char *input,unsigned char *output,const float alpha,const int beta,const unsigned int height,const unsigned int width,const unsigned int channel);
```
[the remark this answers is not in the document]
Comment (r-abishek) [reply src/include/hip/rpp/hip_kernels.h:1075]: The low level function is actually called only brightness for host. Similarly we have another function called contrast. Just wanted to logically match the names for host and hip.

File: src/include/hip/rpp/hip_kernels.h [hunk 8]
```diff
@@ -1672,9 +1663,6 @@ const std::map<std::string, const void*>& funMap1()
                                                         {"gaussian_image_pyramid_pln_batch",reinterpret_cast<const void*>(gaussian_image_pyramid_pln_batch)},
                                                         {"laplacian_image_pyramid_pkd_batch",reinterpret_cast<const void*>(laplacian_image_pyramid_pkd_batch)},
                                                         {"laplacian_image_pyramid_pln_batch",reinterpret_cast<const void*>(laplacian_image_pyramid_pln_batch)},
-                                                        {"ced_pln3_to_pln1_batch",reinterpret_cast<const void*>(ced_pln3_to_pln1_batch)},
```
[the remark this answers is not in the document]
Comment (r-abishek) [reply src/include/hip/rpp/hip_kernels.h:1675]: No, they were duplicate definitions that was causing the STATIC build to not compile properly. For now, the functionality is the same.

File: src/modules/hip/kernel/blend.cpp (hunk 1 above)
Comment (r-abishek) [reply src/modules/hip/kernel/blend.cpp:33]: We'll do the kernel level optimizations subsequent PRs

File: src/modules/hip/kernel/brightness.cpp (hunk 2 above)
Comment (r-abishek) [reply src/modules/hip/kernel/brightness.cpp:68]: We'll do the kernel level optimizations subsequent PRs

File: src/modules/hip/kernel/fog.cpp (hunk 3 above)
Comment (r-abishek) [reply src/modules/hip/kernel/fog.cpp:30]: We'll do the kernel level optimizations subsequent PRs

File: src/modules/hip/kernel/fog.cpp (hunk 4 above)
Comment (r-abishek) [reply src/modules/hip/kernel/fog.cpp:36]: We'll do the kernel level optimizations subsequent PRs

File: src/modules/hip/kernel/snow.cpp (hunk 5 above)
Comment (r-abishek) [reply src/modules/hip/kernel/snow.cpp:130]: We'll do the kernel level optimizations subsequent PRs

File: src/modules/hip/kernel/snow.cpp (hunk 6 above)
Comment (r-abishek) [reply src/modules/hip/kernel/snow.cpp:171]: We'll do the kernel level optimizations subsequent PRs

**Review by rrawther — APPROVED** [review]

**Review by rrawther — APPROVED** [review]
Please make sure all the unit tests pass for this PR

**Review by asalmanp — APPROVED** [review]

**Comment by rrawther** [comment]
@kiritigowda : can we merge this?

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**